tests/unit/utils/test_misc.py

```diff
@@ -50,7 +53,8 @@
             tmp_dir.mkdir(parents=True, exist_ok=True)
             expected = tmp_dir / "app.elf"
             with self.assertRaises(AssertionError) as error:
-                misc.find_main_application(dir_path, device)
+                misc.find_main_application(dir_path, conf.OPTIONAL.BUILD_DIR,
+                                           conf.OPTIONAL.BIN_NAME, device)
```

Comment:
We've cornered an error with the current system when using the `LOAD_MAIN_APP_AS_LIBRARY` option.
This option is used when the `MAIN_APP` (the app from the current repository) is to be loaded as a library. In this case, the `app_path` refers to another app which will be the app started by Speculos. 

This is the case for eth plugins:
- the tests run from the plugin repository, the plugin is the main app
- `app_path` will point into the ETH app binary

This highlights an ambiguous usage of the `APP_DIR` variable. 
- On most cases it is the repository of the current application, and its value is either `.` (default value) or `./app` (Zondax app). This becomes redundant with manifests as it is exactly the `app.build_directory` value.
- On plugins cases it becomes the ETH app binary root directory. This is the buggy case.

We agreed on the following:
- using the qualifier `MAIN_APP` to designate the application which should be run first, instead of designating the application from which repo the tests runs;
- application from which repo the tests runs is designated as `LOCAL_APP`;
- A new optional value `MAIN_APP_DIR` is created, which if not None means that:
  - the `LOCAL_APP` is not the `MAIN_APP` and thus shall be loaded as a library (`LOAD_MAIN_APP_AS_LIBRARY` becomes obsolete)
  - the `APP_DIR` of this foreign `MAIN_APP` is this `MAIN_APP_DIR` directory.

This to our mind will make this part clearer.

An issue remains: without the `MAIN_APP` manifest, it is not possible to know if the `MAIN_APP` is a Rust or C application, and from now inferring the proper binary path is not possible.
As there is no plan to have applications running as libraries of a Rust application, we thought this could be ignored for now, and that we could assume every `MAIN_APP` which are not `LOCAL_APP` are C applications.